tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -329,20 +374,49 @@ private void AssertAuthenticationFails(MongoClientSettings settings)
         {
             return CreateDatabaseUser(client, "admin", userName, password, "readWriteAnyDatabase", mechanisms);
         }
-        
+
         private BsonDocument CreateDatabaseUser(
             MongoClient client,
             string source,
-            string userName, 
-            string password, 
-            string role, 
+            string userName,
+            string password,
+            string role,
             params string[] mechanisms)
         {
             var createUserCommand =
-                $"{{createUser: '{userName}', pwd: '{password}'," 
+                $"{{createUser: '{userName}', pwd: '{password}',"
                 + $"roles: ['{role}'], mechanisms: {mechanisms.ToJson()}}}";
             return client.GetDatabase(source).RunCommand<BsonDocument>(createUserCommand);
         }
-        
+
+        private BsonDocument CreateX509DatabaseUser(MongoClient client, string userName)
+        {
+            var createUserCommand =
+                $"{{ createUser : '{userName}'," +
+                @"   roles : [ { role : 'readWrite', db : 'test' },
+                               { role : 'userAdminAnyDatabase', db : 'admin' } ]
+                  }";
+            return client.GetDatabase("$external").RunCommand<BsonDocument>(createUserCommand);
+        }
+
+        private void DropDatabaseUser(MongoClient client, string userName, string database)
+        {
+            var dropUserCommand = $"{{ dropUser : '{userName}' }}";
+
+            try
+            {
+                client.GetDatabase(database).RunCommand<BsonDocument>(dropUserCommand);
+            }
+            catch (MongoCommandException) // command throws if user doesn't exist
+            {
```

Comment:
Is is it safe to ignore ALL MongoCommandExceptions?

Or do we need to look at the MongoCommandException and only ignore it if the error message (or code) confirms that it's a user does not exist error?

I realize this is only test code so a certain amount of looseness may be tolerated.